examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,335 @@
+/****************************************************************************
+ * Copyright (c) 2017-2023 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+// Example taken from DataTransferKit
+// (https://github.com/ORNL-CEES/DataTransferKit)
+// with MLS resolution from
+// (http://dx.doi.org/10.1016/j.jcp.2015.11.055)
+
+#include <ArborX.hpp>
+#include <Kokkos_Core.hpp>
+
+#include <limits>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+struct RBFWendland_0
+{
+  KOKKOS_INLINE_FUNCTION double operator()(double x)
+  {
+    x /= _radius;
+    return (1. - x) * (1. - x);
+  }
+
+  double _radius;
+};
+
+struct MVPolynomialBasis_3D
+{
+  static constexpr std::size_t size = 4;
+
+  template <typename Double3D>
+  KOKKOS_INLINE_FUNCTION Kokkos::Array<double, size>
+  operator()(Double3D const &p) const
```

Comment:
I think it should be templated on a `Point`. We can assert that the point is 3D (can we extend the basis to other dimensions)? Also, would want to switch away from using `double`.

Reply:
This was taken directly from DTK's code of different polynomial bases. I will strip out the templates and switch `double` to `float` on the example. As I am using `ArborX::Point`, it is 3D.

I think it is possible to generate that code for any number of dimensions and polynomial degree, but it will most likely require a lot of templated code and classes.